tests/end-to-end/api/01-users.js

```diff
@@ -286,6 +315,106 @@ describe('[Users]', function() {
 				})
 				.end(done);
 		});
+
+		it('should return an error when trying update profile and it is not allowed', (done) => {
+			updateSetting('Accounts_AllowUserProfileChange', false)
+				.then(() => {
+					request.post(api('users.update'))
+						.set(credentials)
+						.send({
+							userId: targetUser._id,
+							data: {
+								verified: true
+							}
+						})
+						.expect('Content-Type', 'application/json')
+						.expect(400)
+						.expect((res) => {
+							expect(res.body).to.have.property('success', false);
+						})
+						.end(done);
+				});
+		});
+
+		it('should return an error when trying update username and it is not allowed', (done) => {
+			updateSetting('Accounts_AllowUsernameChange', false)
+				.then(() => {
+					request.post(api('users.update'))
+						.set(credentials)
+						.send({
+							userId: targetUser._id,
+							data: {
+								username: 'fake.name'
+							}
+						})
+						.expect('Content-Type', 'application/json')
+						.expect(400)
+						.expect((res) => {
+							expect(res.body).to.have.property('success', false);
+						})
+						.end(done);
+				});
+		});
+
+		it('should return an error when trying update user real name and it is not allowed', (done) => {
+			updateSetting('Accounts_AllowRealNameChange', false)
+				.then(() => {
+					request.post(api('users.update'))
+						.set(credentials)
+						.send({
+							userId: targetUser._id,
+							data: {
+								name: 'Fake name'
+							}
+						})
+						.expect('Content-Type', 'application/json')
+						.expect(400)
+						.expect((res) => {
+							expect(res.body).to.have.property('success', false);
+						})
+						.end(done);
+				});
+		});
+
+		it('should return an error when trying update user email and it is not allowed', (done) => {
+			updateSetting('Accounts_AllowEmailChange', false)
```

Comment:
I am curious why this test did not find the bug from above. Maybe you need beforeEach to reset settings?